### PR TITLE
adding is.empty function and blank functionality

### LINKED ISSUE #7: Handling Variables with constant values
See image below. Here this is supposed to be a datetime variable, but all the values are blank characters... ie: c("","","","",""). When this happens, the dropdown widget here is misleading/confusing as it prompts the user to input some text, which will have no effect on the filtering process whatsoever. As such, perhaps we limit users to filter variables that are not constant in the dataframe? Or pursue another solution?

image

### PR DESCRIPTION
**Pull request by MayaGans**
- Added a column to our dummy data where every value is `""` [I hope this replicates issue #7?]

- Added function `is.empty` to utils which turns a logical if the input == `""`

- In the character method if you map over the column and all the values are `""` then the output is a blank div with the text "Input is empty" (pictured)

Screen Shot 2020-04-08 at 7 51 49 PM

Closes #7

### CODE REVIEW

File: R/shiny_vector_filter_character.R [hunk 1]
```diff
@@ -19,12 +19,22 @@ shiny_vector_filter.character <- function(data, inputId, ...) {
...
-                          vec = x,
-                          selected = shiny::isolate(input$param) %||% c(),
-                          multiple = TRUE,
-                          width = "100%")
+    
+    if (map(x(), is.empty) %>% reduce(`&`)) {
+      shiny::div(style = "opacity: 0.5;",
+                 p(width = "100%", 
+                   align = "center", 
+                   "Input is blank"))
```
Comment (AARON-CLARK) [inline R/shiny_vector_filter_character.R]: `"Column is blank"))`

Comment (AARON-CLARK) [reply R/shiny_vector_filter_character.R]: Since the users don't know what an "input" is, could we use the word "Column" or "Variable" here?

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
See comment regarding "Input is blank".

File: R/shiny_vector_filter_character.R (hunk 1 above)
Comment (AARON-CLARK) [reply R/shiny_vector_filter_character.R]: Or better yet, "Variable only contains blank values"??

Comment (MayaGans) [reply R/shiny_vector_filter_character.R]: Great idea! I was just focused on getting it working - love that

**Review by AARON-CLARK — APPROVED** [review]
two thumbs up

**Pull request merged**